src/main/java/by/exadel/internship/service/impl/UserTimeSlotServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package by.exadel.internship.service.impl;
+
+
+import by.exadel.internship.dto.time_for_call.UserTimeSlotWithUserDTO;
+import by.exadel.internship.dto.time_for_call.UserTimeSlotWithUserIdDTO;
+import by.exadel.internship.dto.time_for_call.UserTimeSlotDTO;
+import by.exadel.internship.dto.UserDTO;
+import by.exadel.internship.mapper.UserTimeSlotMapper;
+import by.exadel.internship.repository.UserTimeSlotRepository;
+import by.exadel.internship.service.UserTimeSlotService;
+import by.exadel.internship.service.UserService;
+import by.exadel.internship.util.MDCLog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+;
+
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UserTimeSlotServiceImpl implements UserTimeSlotService {
+
+    private final UserTimeSlotRepository userTimeSlotRepository;
+    private final UserService userService;
+    private final UserTimeSlotMapper mapper;
+
+    private static final String SIMPLE_CLASS_NAME = UserTimeSlotService.class.getSimpleName();
+
+
+    @Override
+    public void saveUserTime(UserDTO userDTO) {
```

Comment:
Get the whole user object is excessive in this method. Better to work with List<UserTimeSlotDTO>

Reply:
I asked a question in skype)